blocks/amp-story/amp-story-page.js

```diff
@@ -0,0 +1,93 @@
+const { __ } = wp.i18n;
+const {
+	registerBlockType
+} = wp.blocks;
+const {
+	InspectorControls,
+	InnerBlocks
+} = wp.editor;
+const {
+	TextControl,
+	Notice
+} = wp.components;
+
+const TEMPLATE = [
+	[
+		'amp/amp-story-grid-layer',
+		[],
+		[
+			[
+				'core/paragraph',
+				{
+					placeholder: __( 'Add content to layer.', 'amp' )
+				}
+			]
+		]
+	]
+];
+
+const ALLOWED_BLOCKS = [
+	'amp/amp-story-grid-layer',
+	'amp/amp-story-cta-layer'
+];
+
+/**
+ * Register block.
+ */
+export default registerBlockType(
+	'amp/amp-story-page',
+	{
+		title: __( 'AMP Story Page', 'amp' ),
+		category: 'layout',
+		icon: 'admin-page',
+		attributes: {
+			id: {
+				type: 'string',
+				source: 'attribute',
+				selector: 'amp-story-page',
+				attribute: 'id'
+			}
+		},
+
+		/*
+		 * <amp-story-page>:
+		 *   mandatory_parent: "AMP-STORY"
+		 *   mandatory_min_num_child_tags: 1
+		 *   child_tag_name_oneof: "AMP-ANALYTICS"
+		 *   child_tag_name_oneof: "AMP-PIXEL"
+		 *   child_tag_name_oneof: "AMP-STORY-CTA-LAYER"
+		 *   child_tag_name_oneof: "AMP-STORY-GRID-LAYER"
+		 *
+		 * https://github.com/ampproject/amphtml/blob/87fe1d02f902be97b596b36ec3421592c83d241e/extensions/amp-story/validator-amp-story.protoascii#L146-L171
+		 * */
+
+		// @todo Show error if no ID is supplied.
+		edit( props ) {
+			const { setAttributes } = props;
+			return [
+				<InspectorControls key='inspector'>
+					<TextControl
+						type="text"
+						className="blocks-amp-story-page__id"
+						required={ true }
+						label={ __( 'ID', 'amp' ) }
+						value={ props.attributes.id }
+						onChange={ value => ( setAttributes( { id: value } ) ) }
+					/>
+				</InspectorControls>,
+				! props.attributes.id && (
+					<Notice status="error" isDismissible={ false }>{ __( 'You must supply an ID for the page.', 'amp' ) }</Notice>
+				),
+				<InnerBlocks key='contents' template={ TEMPLATE } allowedBlocks={ ALLOWED_BLOCKS } />
```

Comment:
This should ensure not being able to add `amp-story-page` as a child for `amp-story-page`.

Reply:
@westonruter Just saw that you tried with `allowedBlocks` within https://github.com/Automattic/amp-wp/commit/9cd428e38535c0b48b01b25dc6259a92c1eac2ff and it created an error. Are you getting the error now as well? I'm not seeing the error at this moment, wondering if I should actually.

Reply:
Yes, I had tried it but I was getting an error. However, if you're not getting an error now, then perhaps the cause was fixed in Gutenberg.